app/src/main/java/com/pvp/app/ui/screen/steps/StepViewModel.kt

```diff
@@ -40,22 +39,47 @@ class StepViewModel @Inject constructor(
 
     @RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
     fun updateTodaysSteps() {
+        viewModelScope.launch {
+            val end = LocalDate
+                .now()
+                .plusDays(1)
+                .atStartOfDay(ZoneId.systemDefault())
+                .toInstant()
+            val start = LocalDate
+                .now()
+                .atStartOfDay(ZoneId.systemDefault())
+                .toInstant()
+
+            _stepsCount.value = service.aggregateSteps(
+                start = start,
+                end = end
+            )
+        }
+    }
+
+    fun getExercises() {
         viewModelScope.launch {
             val end = Instant.now()
-            val start = LocalDate.now().atStartOfDay(ZoneId.systemDefault()).toInstant()
-
-            try {
-                val records: List<StepsRecord> = service.readActivityData(
-                    record = StepsRecord::class,
-                    start = start,
-                    end = end
-                )
-
-                _stepsCount.value = records.sumOf { it.count.toInt() }
-            } catch (e: Exception) {
-                // Exception should be handled by showing a Toast
-                e.printStackTrace()
+            val start = LocalDate
+                .now()
+                .minusDays(29)
+                .atStartOfDay(ZoneId.systemDefault())
+                .toInstant()
+
+            val records: List<ExerciseSessionRecord> = service.readActivityData(
```

Comment:
Explicit type may be redundant, if readActivityData returns `List<T>`, where `T` is record of `KClass<T>`.